src/main/java/org/apache/commons/lang3/builder/ToStringStyle.java

```diff
@@ -635,6 +635,14 @@ protected void appendDetail(final StringBuffer buffer, final String fieldName, f
      *  {@code toString}, not {@code null}
      */
     protected void appendDetail(final StringBuffer buffer, final String fieldName, final Collection<?> coll) {
+        if (coll != null && !coll.isEmpty()) {
+            coll.stream().findFirst()
+                    .map(Object::getClass)
+                    .filter(Class::isEnum)
```

Comment:
This will restrict only enums to be appended in a collection, a list of other types (e.g. Collection<Integer> or Collection<String>) will be ignored. This filter should be removed.

Reply:
We should avoid using streams here IMO, it will likely make performance worse in what can already be slow when reflection is involved.

Reply:
@garydgregory @swarajsaaj I added some code following your comments. Please take a look. Thank you!

Reply:
This should fix the case I mentioned earlier. Seems good.

Reply:
@swarajsaaj, I added test case for it